lib/destination/dwh.go

```diff
@@ -35,3 +37,37 @@ type Baseline interface {
 	IsRetryableError(err error) bool
 	IdentifierFor(topicConfig kafkalib.TopicConfig, table string) types.TableIdentifier
 }
+
+// ExecStatements executes one or more statements against a [DataWarehouse].
+// If there is more than one statement, the statements will be executed inside of a transaction.
+func ExecStatements(dwh DataWarehouse, statements []string) error {
+	switch len(statements) {
+	case 0:
+		return fmt.Errorf("statements is empty")
+	case 1:
+		slog.Debug("Executing...", slog.String("query", statements[0]))
+		if _, err := dwh.Exec(statements[0]); err != nil {
+			return fmt.Errorf("failed to execute statement: %w", err)
+		}
+
+		return nil
+	default:
+		tx, err := dwh.Begin()
```

Comment:
What about rolling back tx if it failed?

Reply:
I can add that, I was just duplicating what we were already doing for Redshift merges https://github.com/artie-labs/transfer/blob/e275e192b8a886f5d8c4498da964ce37b85dd062/clients/shared/merge.go#L147-L168

Reply:
Yeah, we should add it. We're also inconsistent about rolling it back.